### PR TITLE
Datatable CRUD feature created

### PR DESCRIPTION
**Pull request by teamdraftbox**
Datatable module created with CRUD capabilities
1. Create and show list API's created with filter capabilities, with data being injected from pg Database
2. Delete API's to delete record.
3. Sonar clean up first cut done for the funtionality

### CODE REVIEW

File: src/main/java/com/strandls/dataTable/ApiConstants.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+public class ApiConstants {
+	
+	/**
+	 * since all the class members are static we need to add a private constructor 
+	 */
+	private ApiConstants() {
+		super();
+	}
+
+	// versioning
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/ApiConstants.java:20]: code formating required

File: src/main/java/com/strandls/dataTable/DataTableServeletContextListener.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+				Map<String, String> props = new HashMap<String, String>();
+				props.put("javax.ws.rs.Application", ApplicationConfig.class.getName());
+				props.put("jersey.config.server.provider.packages", "com");
+				props.put("jersey.config.server.wadl.disableWadl", "true");
+
+				bind(ServletContainer.class).in(Scopes.SINGLETON);
+				bind(UserServiceApi.class).in(Scopes.SINGLETON);
+				bind(ResourceServicesApi.class).in(Scopes.SINGLETON);
+				bind(ActivitySerivceApi.class).in(Scopes.SINGLETON);
+				bind(WKTWriter.class).in(Scopes.SINGLETON);
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/DataTableServeletContextListener.java:90]: to which instance its getting binded ?

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+
+		<!-- https://mvnrepository.com/artifact/com.zaxxer/HikariCP -->
+		<dependency>
+			<groupId>com.zaxxer</groupId>
+			<artifactId>HikariCP</artifactId>
+			<version>3.4.2</version>
+		</dependency>
+
+		<dependency>
+			<groupId>com.strandls.biodiv</groupId>
```
Comment (akrudra) [inline pom.xml:249]: repeadted twice auth utils

File: src/main/java/com/strandls/dataTable/Headers.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+import com.strandls.resource.controllers.ResourceServicesApi;
+import com.strandls.user.controller.UserServiceApi;
+
+/**
+ * 
+ * @author vishnu
+ *
+ */
+public class Headers {
+
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/Headers.java:15]: class not binded to context listner

File: src/main/java/com/strandls/dataTable/dao/DataSetDAO.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+		return entity;
+	}
+
+	public Dataset findDataSetByTitle(String title) {
+		Session session = sessionFactory.openSession();
+		String qry = "select * from dataset1 where is_deleted = false and title = :keyword limit 1";
+
+		Dataset result = null;
+		try {
+			Query<Dataset> query = session.createNativeQuery(qry, Dataset.class).setParameter("keyword", title);
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/dao/DataSetDAO.java:44]: test this query, its running or not

File: src/main/java/com/strandls/dataTable/dao/DataTableDAO.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		List<DataTable> observationList = new ArrayList<DataTable>();
+		String hql = "from DataTable where is_deleted = false  order by :orderBy desc";
+		try {
+			Query query = session.createQuery(hql);
+			query.setParameter("orderBy", orderBy);
+			query.setFirstResult(offset);
+			if (limit != null) {
+				query.setMaxResults(limit);
+			}
+			observationList = query.list();
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/dao/DataTableDAO.java:55]: query.list or query.getresultList ?????

File: src/main/java/com/strandls/dataTable/pojo/DataTableListMapping.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
+package com.strandls.dataTable.pojo;
+
+import java.util.Date;
+
+import com.strandls.user.pojo.UserIbp;
+
+public class DataTableListMapping {
+
+	
+	private Long id;
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/pojo/DataTableListMapping.java:10]: format your code and add a default constructor

File: src/main/java/com/strandls/dataTable/service/impl/DataTableServiceModule.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+ * 
+ * @author vishnu
+ *
+ */
+public class DataTableServiceModule extends AbstractModule{
+
+	@Override
+	protected void configure() 	{
+		
+		bind(DataTableService.class).to(DataTableServiceImpl.class).in(Scopes.SINGLETON);
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/service/impl/DataTableServiceModule.java:17]: format the code, use proper indentaion in all the modules

File: src/main/java/com/strandls/dataTable/util/LogActivities.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+			ActivityLoggingData activityLogging = new ActivityLoggingData();
+			activityLogging.setActivityDescription(activityDescription);
+			activityLogging.setActivityId(activityId);
+			activityLogging.setActivityType(activityType);
+			activityLogging.setRootObjectId(rootObjectId);
+			activityLogging.setRootObjectType(rootObjectType);
+			activityLogging.setSubRootObjectId(subRootObjectId);
+			activityLogging.setMailData(mailData);
+			activityService = headers.addActivityHeaders(activityService, authHeader);
+			activityService.logActivity(activityLogging);
```
Comment (akrudra) [inline src/main/java/com/strandls/dataTable/util/LogActivities.java:41]: Seems this is observation activity, shouldnt it be data table activity

**Review by akrudra — CHANGES_REQUESTED** [review]

File: src/main/java/com/strandls/dataTable/pojo/DataTableListMapping.java (hunk 7 above)
Comment (teamdraftbox) [reply src/main/java/com/strandls/dataTable/pojo/DataTableListMapping.java:9]: added default constructor

File: src/main/java/com/strandls/dataTable/Headers.java (hunk 4 above)
Comment (teamdraftbox) [reply src/main/java/com/strandls/dataTable/Headers.java:15]: binded

File: src/main/java/com/strandls/dataTable/util/LogActivities.java (hunk 9 above)
Comment (teamdraftbox) [reply src/main/java/com/strandls/dataTable/util/LogActivities.java]: changed with datatable activity

**Review by akrudra — APPROVED** [review]

**Pull request merged**